### PR TITLE
update required node version [breaking]

### PR DESCRIPTION
**Pull request by janpaepke**
In accordance with the discussion [here](https://github.com/mollie/mollie-api-node/pull/338), this raises the minimum node version for the package to node 14.

It also updates some github actions versions (which should have no effect).

The CI is now also configured to run on node 20.

This update will require the repo settings to be updated to expect the correct checks.

### CODE REVIEW

**Review by fjbender — APPROVED** [review]

**Comment by fjbender** [comment]
I adjusted the branch protection rules to reflect the pipeline changes.

Do you want another review, maybe from @Pimm ?

**Comment by maria-swierblewska** [comment]
Hi @Pimm is it possible for you to review this change?

**Comment by maria-swierblewska** [comment]
Hi @janpaepke this update is showing up as a vulnerability for one of our important integrations, could we skip a second review and merge together with https://github.com/mollie/mollie-api-node/pull/338#issuecomment-1891541701?

**Review by edorivai — APPROVED** [review]
Standing in for @Pimm as he's out sick at the moment. This PR looks good. Changes seem safe; only changing Node versions for test runs. Double checked in test output that all steps are using the intended Node version.

This one is good to go!

**Pull request merged**
